File: Data Structures & Algorithms/two-integer-sum-ii/submission-2.js
class Solution {
    /**
     * @param {number[]} numbers
     * @param {number} target
     * @return {number[]}
     */
    twoSum(numbers, target) {
        for (let i = 0; i < numbers.length; i++){
            let l = i + 1, r = numbers.length - 1;
            let tmp = target - numbers[i]
            while (l <= r){
                let mid = l + Math.floor((r - l) / 2);
                let val = numbers[mid]
                if (val === tmp){
                    return [i + 1, mid + 1]
                } else if (val > tmp){
                    r = mid - 1;
                } else {
                    l = mid + 1
                }
            }
        }
        return [];
    }
}

File: Data Structures & Algorithms/remove-element/submission-2.py
class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        tmp = []

        for n in nums:
            if val == n:
                continue
            tmp.append(n)
        for i in range(len(tmp)):
            nums[i] = tmp[i]
        return len(tmp)